src/cls/lock/cls_lock.cc

```diff
@@ -106,6 +106,10 @@ static int write_lock(cls_method_context_t hctx, const string& name, const lock_
   string key = LOCK_PREFIX;
   key.append(name);
 
+  if (lock.lockers.empty()) {
+    return cls_rmxattr(hctx, key.c_str());
```

Comment:
does this not change the behavior of the `list_locks()` tests?

Reply:
No, it never tested for this. Though I added a check to make it do so.

Reply:
we run these cls tests in the upgrade suites ([pacific-x/parallel/workload/rados_api.yaml](https://github.com/ceph/ceph/blob/b0615fd/qa/suites/upgrade/pacific-x/parallel/workload/rados_api.yaml#L11) and [quincy-x/parallel/workload/rados_api.yaml](https://github.com/ceph/ceph/blob/b0615fd/qa/suites/upgrade/quincy-x/parallel/workload/rados_api.yaml#L11)), so i was concerned that we might introduce a failure there. i think it would actually make life easier *not* to add a test for it